File: punto.java
package punto;

public class punto {
    private double x;
    private double y;

    public punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double[] coordCartesianas() {
        return new double[]{x, y};
    }

    public double[] coordPolares() {
        double r = Math.sqrt(x * x + y * y);
        double theta = Math.atan2(y, x);
        return new double[]{r, theta};
    }

    @Override
    public String toString() {
        return "Punto(" + x + ", " + y + ")";
    }

    public static void main(String[] args) {
        punto p = new punto(3, 4);
        System.out.println(p);  // Output: Punto(3.0, 4.0)
        System.out.println(java.util.Arrays.toString(p.coordCartesianas())); // Output: [3.0, 4.0]
        System.out.println(java.util.Arrays.toString(p.coordPolares())); // Output: [5.0, 0.93...]
    }
}

File: punto.py
import math

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return self.x, self.y

    def coord_polares(self):
        r = math.sqrt(self.x**2 + self.y**2)
        theta = math.atan2(self.y, self.x)
        return r, theta

    def __str__(self):
        return f"Punto({self.x}, {self.y})"

# Ejemplo de uso
p = Punto(3, 4)
print(p)  # Output: Punto(3, 4)
print(p.coord_cartesianas())  # Output: (3, 4)
print(p.coord_polares())  # Output: (5.0, 0.93...)
